src/server.c

```diff
@@ -1126,6 +1110,40 @@ void clientsCron(void) {
     }
 }
 
+/* A periodic timer that performs client maintenance.
+ * This cron task follows the following rules:
+ *  - All clients need to be checked (at least) once per second
+ *  - To manage latency, we don't check more than MAX_CLIENTS_PER_CLOCK_TICK at a time
+ *  - The minimum rate will be defined by server.hz
+ *  - At least CLIENTS_CRON_MIN_ITERATIONS will be performed each cycle
+ */
+#define CLIENTS_CRON_MIN_ITERATIONS 5
+long long clientsTimerProc(struct aeEventLoop *eventLoop, long long id, void *clientData) {
+    UNUSED(eventLoop);
+    UNUSED(id);
+    UNUSED(clientData);
+
+    int numclients = listLength(server.clients);
+    int clients_this_cycle = numclients / server.hz; /* Initial computation based on standard hz */
+    int delay_ms;
+
+    if (clients_this_cycle < CLIENTS_CRON_MIN_ITERATIONS) {
+        clients_this_cycle = min(numclients, CLIENTS_CRON_MIN_ITERATIONS);
+    }
+
+    if (clients_this_cycle > MAX_CLIENTS_PER_CLOCK_TICK) {
+        clients_this_cycle = MAX_CLIENTS_PER_CLOCK_TICK;
+        float required_hz = (float)numclients / MAX_CLIENTS_PER_CLOCK_TICK;
```

Comment:
we need check `CONFIG_MAX_HZ`, ensure the `required_hz` would not exceed `CONFIG_MAX_HZ`.

Reply:
```suggestion
        float required_hz = (float)numclients / MAX_CLIENTS_PER_CLOCK_TICK;
        if (required_hz > CONFIG_MAX_HZ) required_hz = CONFIG_MAX_HZ
```